File: src/main/kotlin/com/tollwood/rechnungen/Employee.kt
package com.tollwood.rechnungen

import javax.persistence.*

@Entity(name="EMPLOYEE")
data class Employee(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        val id: Long,
        val employeeNo: String,
        val firstName: String,
        val lastName: String,
        @Embedded
        val address: Address,
        val phone: String,
        val email: String,
        val taxNo: String
)

File: src/main/webapp/src/employees/Employee.ts
import Link from "../common/Links";
import {Address} from "../common/Address";
import BankDetails from "./BankDetails";

export default class Employee {
    id?: bigint;
    firstName: string = "";
    lastName: string = "";
    taxIdent: string = "";
    email: string = "";
    phone: string = "";
    technicianId: string = "";
    address: Address = new Address();
    bankDetails: BankDetails = new BankDetails();
    _links: {self?: Link} = {};
}